toast-tk-web-agent/src/main/java/io/toast/tk/agent/ui/provider/PropertiesProvider.java

```diff
@@ -0,0 +1,67 @@
+package io.toast.tk.agent.ui.provider;
+
+import com.google.inject.Inject;
+import com.google.inject.Provider;
+import io.toast.tk.agent.config.AgentConfig;
+import io.toast.tk.agent.config.AgentConfigProvider;
+import io.toast.tk.agent.ui.PropertiesHolder;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
+
+public class PropertiesProvider implements Provider<PropertiesHolder> {
+
+    private final AgentConfigProvider configProvider;
+    private File toastWebPropertiesFile;
+    private static final Logger LOG = LogManager.getLogger(PropertiesProvider.class);
+
+    @Inject
+    public PropertiesProvider(AgentConfigProvider configProvider) throws IOException {
+        this.configProvider = configProvider;
+        this.toastWebPropertiesFile = new File(AgentConfig.TOAST_PROPERTIES_FILE);
+        if(!Files.exists(Paths.get(AgentConfig.getToastHome()))){
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Replace this with a call to the "toFile().exists()" method [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS3725)